### PR TITLE
Fix the highlighted gene ID in Gene tree

### PR DESCRIPTION
**Pull request by imransl**
## Description

The highlighted gene ID is not displayed correctly in the Highlighted genes info box. This happens when the orthologue gene ID does not have a display name. The reason behind this was that the code checked if the display name of the orthologue exists and when it fails would display the stable ID of the orthologue instead of the current gene.

The changes in this PR will show the stable ID of the current gene when this check fails. It would be ideal to show the display name instead, but I couldn't figure out how to do it as everything I tried ended up breaking the code.

**Sandbox link:** http://wp-np2-1e.ebi.ac.uk:8330/Homo_sapiens/Gene/Compara_Tree?anc=46819155;db=core;g=ENSG00000133392;g1=ENSEEUG00000000659;r=16:15703135-15857028

**For comparison the staging link is:** http://staging.ensembl.org/Homo_sapiens/Gene/Compara_Tree?anc=46819155;db=core;g=ENSG00000133392;g1=ENSEEUG00000000659;r=16:15703135-15857028

## Views affected

Gene tree -> Highlighted gene info box

## Related JIRA Issues (EBI developers only)

[ENSWEB-6591](https://www.ebi.ac.uk/panda/jira/browse/ENSWEB-6591)

### CODE REVIEW

File: modules/EnsEMBL/Web/Component/Gene/ComparaTree.pm [hunk 1]
```diff
@@ -181,19 +181,25 @@ sub content {
 
   # store g1 param in a different param as $highlight_gene can be undef if highlighting is disabled
   my $gene_to_highlight = $hub->param('g1');
+  my $current_gene = $hub->param('g'); # stable ID of current gene
```
Comment (ens-ap5) [inline modules/EnsEMBL/Web/Component/Gene/ComparaTree.pm:184]: Did you try $self->object->gene->display_xref->db_display_name?

You need to check for the display_xref, something like the code here: https://github.com/Ensembl/ensembl-webcode/blob/master/modules/EnsEMBL/Web/Component/Gene/GeneSummary.pm#L51

**Review by ens-ap5 — CHANGES_REQUESTED** [review]
I think we can get the display name of the current gene, since we're on a gene page. I can go through it with you if you're not sure.

File: modules/EnsEMBL/Web/Component/Gene/ComparaTree.pm (hunk 1 above)
Comment (imransl) [reply modules/EnsEMBL/Web/Component/Gene/ComparaTree.pm:184]: @ens-ap5 Thanks for the suggestion. using `display_xref->display_id` worked.

**Review by ens-ap5 — CHANGES_REQUESTED** [review]
Judging by other parts of the code, you need to check if there is a display_xref object before calling a method on it. This code will work in human but may throw an error in less well-annotated species.

**Comment by imransl** [comment]
> Judging by other parts of the code, you need to check if there is a display_xref object before calling a method on it. This code will work in human but may throw an error in less well-annotated species.

@ens-ap5 I'm now checking if `display_xref` exists and assign the value of `g` param if it doesn't. Is there a better way to do this?

**Review by ens-st3 — COMMENTED** [review]
This is a slightly different bug, but very much related. When the focus gene doesn't have a display name, then the info box shows the display name of the gene from the target species, eg http://wp-np2-1e.ebi.ac.uk:8330/Salvator_merianae/Gene/Compara_Tree?anc=46819267;db=core;g=ENSSMRG00000001859;g1=ENSBBBG00000015773;r=QVOM01000133.1:49351-65331
The display is correct when neither have a display name, eg http://wp-np2-1e.ebi.ac.uk:8330/Salvator_merianae/Gene/Compara_Tree?anc=46819384;db=core;g=ENSSMRG00000001859;g1=ENSUAMG00000025427;r=QVOM01000133.1:49351-65331

**Review by ens-ap5 — APPROVED** [review]
Looks good!

**Pull request merged**